pkg/operator/latencyprofilecontroller/controller.go

```diff
@@ -0,0 +1,212 @@
+	revisionMap := map[int32]struct{}{}
+	uniqueRevisions := []int32{}
+	for _, nodeStatus := range operatorStatus.NodeStatuses {
+		revision := nodeStatus.CurrentRevision
+		if _, ok := revisionMap[revision]; !ok {
+			revisionMap[revision] = struct{}{}
+			uniqueRevisions = append(uniqueRevisions, revision)
+		}
+	}
+
+	// Collect the current latency profile
+	desiredControllerManagerArgumentVals := map[string]string{}
+	configNodeObj, err := c.nodeLister.Get("cluster")
+	if err != nil {
+		return err
+	}
+
+	switch {
+	case configNodeObj.Spec.WorkerLatencyProfile == configv1.DefaultUpdateDefaultReaction:
+		desiredControllerManagerArgumentVals[NodeMonitorGracePeriodArgument] = strconv.Itoa(int(configv1.DefaultNodeMonitorGracePeriod))
+	case configNodeObj.Spec.WorkerLatencyProfile == configv1.DefaultUpdateDefaultReaction:
+		desiredControllerManagerArgumentVals[NodeMonitorGracePeriodArgument] = strconv.Itoa(int(configv1.MediumNodeMonitorGracePeriod))
+	case configNodeObj.Spec.WorkerLatencyProfile == configv1.DefaultUpdateDefaultReaction:
+		desiredControllerManagerArgumentVals[NodeMonitorGracePeriodArgument] = strconv.Itoa(int(configv1.LowNodeMonitorGracePeriod))
+	}
+	// do nothing if workerLatencyProfile is not set
+	if configNodeObj.Spec.WorkerLatencyProfile == "" {
+		// TODO: in case worker latency profile is transitioned
+		// from Default/Medium/Low to ""
+		// check that the required extendedArgs should vanish
+		newCondition := configv1.WorkerLatencyStatusCondition{
+			Owner:              configv1.KubeControllerManagerOperator,
+			Status:             configv1.ConditionUnknown,
+			LastTransitionTime: metav1.Now(),
+			Message:            "Worker latency profile not set",
+		}
+		c.updateConfigNodeStatus(configNodeObj, &newCondition)
+	}
+
+	// For each revision, check that the configmap for that revision have
+	// correct argument values or not
+	revisionsHaveSynced := true
+	for _, revision := range uniqueRevisions {
+		configMapNameWithRevision := fmt.Sprintf("%s-%d", ControllerManagerConfigMapName, revision)
+		configMap, err := c.configMapClient.ConfigMaps(operatorclient.TargetNamespace).Get(ctx, configMapNameWithRevision, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		matches := true
+		for arg, val := range desiredControllerManagerArgumentVals {
+			match, err := configMatchesControllerManagerArgument(configMap, arg, val)
+			if err != nil {
+				return err
+			}
+			matches = matches && match
+		}
+		if !matches {
+			revisionsHaveSynced = false
+			break
+		}
+	}
+
+	if revisionsHaveSynced {
+		// update config node status
+		newCondition := configv1.WorkerLatencyStatusCondition{
+			Owner:              configv1.KubeControllerManagerOperator,
+			Type:               configv1.WorkerLatencyProfileComplete,
+			Status:             configv1.ConditionTrue,
+			LastTransitionTime: metav1.Now(),
+			Reason:             "Changes to worker latency profile",
+			Message:            "Revisions have synced",
```

Comment:
IMHO, a vast majority of the implementation was inspired from [this code fragment in cluster-kube-apiserver-operator](https://github.com/openshift/cluster-kube-apiserver-operator/blob/10569df18e9577663802c0df9861bce540b16e2b/pkg/operator/boundsatokensignercontroller/controller.go#L257-L292) where they try to achieve something similar to check if the observed config have the required values or not.

- We could assume that this should be able to take care of the fact that more/less irrespective of different revision rollout(s) by various config observers the worker latency profile status stays up to date via checks on the given node revision if the required extendedArguments(s) in the observedConfig of the currently rolled revision per node have the required arg value pairs present or not.

- Also, in case nodes are transitioning from revision x -> x+1 due to worker latency profile update we'd want to status to KCMProgressing, but again if another revision x+1 -> x+2 is triggered by some other config observer / the alike we'd still want to stick to KCMComplete throughout the duration of the new rollout since the latency profile args weren't essentially updated.

Reply:
Thanks for the explanation. So if I understand it correctly we do not care if KCM-o is transitioning. Just to check if the current revision contains what we want. So we can ignore targetRevision since we know what should be the final state.

But in a case somebody overrides the KCM operator config to include custom value for `node-monitor-grace-period`. We would never reconcile to what we want (override has a preference). 

I feel it would be nicer to say that we are not progressing but am okay to tell that the operator is trying forever as well... The cluster is in that case in unsupported mode anyway.

Reply:
Kept it at 5 minutes for the resync interval! :)

Reply:
Thanks, I think we can always document this behavior within Worker Latency Profile documentation to make sure users don't touch these sensitive args in the resource through UnsupportedConfigOverrides in case they want the latency profiles on the cluster to behave as expected!